dalesbred/src/main/java/org/dalesbred/internal/instantiation/SqlArrayConversion.java

```diff
@@ -37,17 +37,17 @@
 import java.util.List;
 import java.util.function.Function;
 
+import static java.util.Objects.requireNonNull;
+
 final class SqlArrayConversion {
 
     private final @NotNull Type elementType;
 
     private final @NotNull InstantiatorProvider instantiatorRegistry;
 
-    private SqlArrayConversion(@NotNull Type elementType,
-                               @NotNull InstantiatorProvider instantiatorRegistry) {
-
+    private SqlArrayConversion(@NotNull Type elementType, @NotNull InstantiatorProvider instantiatorRegistry) {
         this.elementType = elementType;
-        this.instantiatorRegistry = instantiatorRegistry;
+        this.instantiatorRegistry = requireNonNull(instantiatorRegistry);
```

Comment:
I haven't bothered with `requireNonNull` in cases where the API is internal and there is a `@NotNull` annotation in place. Anyway, if it's added for one parameter, why not the other?

So I guess this change is left over from some work and is not really needed.